### PR TITLE
[JENKINS-50616] - Add org.jruby.RubyNil to the whitelist

### PR DESCRIPTION
**Pull request by oleg-nenashev**
See [JENKINS-50616](https://issues.jenkins-ci.org/browse/JENKINS-50616). It is a workaround, because we're stuck in https://github.com/jenkinsci/ruby-runtime-plugin/pull/6 due to the lack of response from plugin maintainers.

### Proposed changelog entries

* Bug: Add org.jruby.RubyNil to the built-in whitelist ( JEP-200 regression in the CI Skip plugin)
* ...

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
@reviewbybees

File: core/src/main/resources/jenkins/security/whitelisted-classes.txt [hunk 1]
```diff
@@ -174,14 +174,21 @@ org.jenkinsci.jruby.JRubyMapper$DynamicProxy
 org.jenkinsci.lib.xtrigger.XTriggerCause
 org.jenkinsci.lib.xtrigger.XTriggerCauseAction
 
-# TODO remove (also XStream2.BlacklistedTypesConverter.JRUBY_PROXY) when https://github.com/jenkinsci/ruby-runtime-plugin/pull/5 is widely adopted
+# TODO remove (also XStream2.BlacklistedTypesConverter.JRUBY_PROXY) when Ruby Runtime is fixed
+# Related PRs:
+#   - https://github.com/jenkinsci/ruby-runtime-plugin/pull/5,
+#   - https://github.com/jenkinsci/ruby-runtime-plugin/pull/6
+#
+# oleg:nenashev in PR#6 we are trying to get help from last maintainers due to the plugin codebase splitbrain.
```
Comment (jglick) [inline core/src/main/resources/jenkins/security/whitelisted-classes.txt]: `oleg-nenashev`?

**Review by jglick — APPROVED** [review]

File: core/src/main/resources/jenkins/security/whitelisted-classes.txt (hunk 1 above)
Comment (oleg-nenashev) [reply core/src/main/resources/jenkins/security/whitelisted-classes.txt]: I have different symbols in GitHub and LDAP accounts, so whatever works :D Will fix it

**Comment by oleg-nenashev** [comment]
@reviewbybees done

**Comment by oleg-nenashev** [comment]
@jenkinsci/code-reviewers Are you fine with me merging it? I would like to propose a backport to 2.107.3 RC today, and it would be great to get it merged to the master first.

**Review by olivergondza — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
@olivergondza would it be possible to also cherry pick this fix to 2.107.3 LTS?

**Comment by olivergondza** [comment]
@oleg-nenashev, done!

**Comment by oleg-nenashev** [comment]
Thanks a lot!

**Pull request merged**